### PR TITLE
Trying to collect info about ping test failures in macOS 10.12

### PR DESCRIPTION
**Pull request by pjanotti**
See #26358

### CODE REVIEW

**Comment by pjanotti** [comment]
Trying macOS outer loop to see if that issue happens...

@dotnet-bot test Outerloop OSX x64 Debug Build 
@dotnet-bot test Outerloop OSX x64 Release Build

**Comment by pjanotti** [comment]
No repro, giving one more try...

@dotnet-bot test Outerloop OSX x64 Debug Build
@dotnet-bot test Outerloop OSX x64 Release Build

**Comment by pjanotti** [comment]
no luck in getting a repro to capture stderr and exit code.

**Comment by pjanotti** [comment]
@Priya91 @danmosemsft I changed the code to capture exit code and stderr in case of failure. I tried to get a repro without merging but no luck. Unless you know something else about this issue #26358 I'm planning to merge it and let it run so I can see the exact exit code and stderr in case of failures.

**Comment by pjanotti** [comment]
The test is in inner loop but trying outer loop too, since this may be influenced by stress

@dotnet-bot test Outerloop OSX x64 Debug Build
@dotnet-bot test Outerloop OSX x64 Release Build
@dotnet-bot test Outerloop Linux x64 Debug Build
@dotnet-bot test Outerloop Linux x64 Release Build

File: src/System.Net.Ping/tests/FunctionalTests/UnixPingUtilityTests.cs [hunk 1]
```diff
@@ -33,21 +34,46 @@ public class UnixPingUtilityTests
...
+                throw new Exception(
+                    $"[{utilityPath} {arguments}] process did not exit in {TestSettings.PingTimeout} ms.\nStdOut:[{pingOutput}]\nStdErr:[{stdErr}]");
             }
 
+            // Ensure standard output and error are flushed
+            p.WaitForExit();
+
+            pingOutput = string.Join("\n", stdOutLines);
+            var exitCode = p.ExitCode;
+            if (exitCode != 0)
```
Comment (danmosemsft) [inline src/System.Net.Ping/tests/FunctionalTests/UnixPingUtilityTests.cs]: There can be stderr output with a zero exit code.

**Comment by danmosemsft** [comment]
Fine by me but I am skeptical this info will help. It will show ping doing the wrong thing on OSX10.12. I am inclined to suggest we just disable ping tests on that version, or make them tolerant this specific failure mode.

[Looks like ](https://netmarketshare.com/operating-system-market-share.aspx?options=%7B%22filter%22%3A%7B%22%24and%22%3A%5B%7B%22deviceType%22%3A%7B%22%24in%22%3A%5B%22Desktop%2Flaptop%22%5D%7D%7D%2C%7B%22platform%22%3A%7B%22%24in%22%3A%5B%22Mac%20OS%22%5D%7D%7D%5D%7D%2C%22dateLabel%22%3A%22Custom%22%2C%22attributes%22%3A%22share%22%2C%22group%22%3A%22platform%2CplatformVersion%22%2C%22sort%22%3A%7B%22share%22%3A-1%7D%2C%22id%22%3A%22platformsDesktop%22%2C%22dateInterval%22%3A%22Monthly%22%2C%22dateStart%22%3A%222017-10%22%2C%22dateEnd%22%3A%222018-01%22%2C%22hiddenSeries%22%3A%7B%7D%2C%22segments%22%3A%22-1000%22%7D )10.13 is approaching half the market now.

File: src/System.Net.Ping/tests/FunctionalTests/UnixPingUtilityTests.cs (hunk 1 above)
Comment (pjanotti) [reply src/System.Net.Ping/tests/FunctionalTests/UnixPingUtilityTests.cs]: Good point, but if it matters to us parse will fail and then it is added to the exception.

**Comment by pjanotti** [comment]
@danmosemsft you're likely right but I'm curious to see it 😸

**Pull request merged**